spark/src/main/java/io/ddf/spark/content/SchemaHandler.java

```diff
@@ -106,5 +127,260 @@ public static Schema getSchemaFromDataFrame(DataFrame rdd) {
     List<Object> listValues = FactorIndexer.getFactorMapForColumn(this.getDDF(), columnName).values();
     return listValues;
   }
+
+  /**
+   * applySchema(Schema newSchema) will do apply change column type from current to new data type. This function return
+   * new DDF and also statistic when its applied new schema
+   *
+   * @param schema will use to apply for current DDF
+   * @return Tuple2<SparkDDF,ApplySchemaStatistic> +SparkDDF: new sparkddf which applied new schema already
+   * +ApplySchemaStatistic:  applied schema statistic @see ApplySchemaStatistic
+   */
+  public Tuple2<SparkDDF, ApplySchemaStatistic> applySchema(Schema schema) throws DDFException {
+    long startApplySchema = System.currentTimeMillis();
+    // Keep info to Convert column at <pos> to <ColumnType>
+    List<Tuple3<Integer, String, Schema.ColumnType>> listConvertColumns = getListConvertColumn(this.getSchema(), schema);
+
+    try {
+      SparkDDFManager manager = (SparkDDFManager) this.getManager();
+//      SparkContext sparkContext = manager.getSparkContext();
+      Accumulator<ApplySchemaStatistic> statisticAccumulator = manager.getSparkContext().accumulator(new ApplySchemaStatistic(), ApplySchemaStatisticParam.getInstance());
+
+      JavaRDD<Row> rdd = ((SparkDDF) this.getDDF()).getRDD(Row.class).toJavaRDD();
+      JavaRDD<Row> filteredRdd = rdd.filter(row -> convertRowType(row, listConvertColumns, statisticAccumulator));
```

Comment:
@Huandao0812 @zkidkid I think the pain point here is that we not only want to do the type conversion, but also wants to get statistics for the conversion. For example, how many rows fails. Thus we need things like counter. But i'm also concerned about the performance, although both may be O(1), but still performance can be different. @zkidkid Do you think we should do a small benchmark using spark to check the performance of scanning row by row and dataframe's cast?